github-copilot-sdk: new Resource() breaks with @opentelemetry/resources v2.x

## Description

The `github-copilot-sdk/src/telemetry.ts` file uses `new Resource({...})` to construct an OpenTelemetry resource. This breaks when `@opentelemetry/resources` v2.x is installed because `Resource` was changed from a class to a **type-only export** in the OTel JS SDK 2.0 release.

## Error

```
src/telemetry.ts - error TS2693: 'Resource' only refers to a type, but is being used as a value here.

  const resource = new Resource({
                        ~~~~~~~~
```

## Root Cause

`@opentelemetry/resources` v2.x (e.g. 2.6.1) introduced a breaking change: `Resource` is no longer a class and cannot be instantiated with `new`. The current dependencies resolve to v2.x, triggering this build failure.

## Affected Code

https://github.com/dynatrace-oss/dynatrace-ai-agent-instrumentation-examples/blob/main/github-copilot-sdk/src/telemetry.ts

```ts
import { Resource } from "@opentelemetry/resources";
// ...
const resource = new Resource({
  [ATTR_SERVICE_NAME]: serviceName,
});
```

## Fix

Replace the class constructor with the new `resourceFromAttributes()` factory function:

```ts
import { resourceFromAttributes } from "@opentelemetry/resources";
// ...
const resource = resourceFromAttributes({
  [ATTR_SERVICE_NAME]: serviceName,
});
```

## Blast Radius

Any other examples in this repo that use `new Resource(...)`, `Resource.default()`, `Resource.empty()`, or `resource.merge()` will also break with `@opentelemetry/resources` v2.x. The v2.x equivalents are:

| v1.x (broken) | v2.x (replacement) |
|---|---|
| `new Resource({ ... })` | `resourceFromAttributes({ ... })` |
| `Resource.default()` | `defaultResource()` |
| `Resource.empty()` | `emptyResource()` |
| `resource.merge(other)` | `mergeResources(resource, other)` |

## References

- [OTel JS SDK 2.0 migration guide](https://github.com/open-telemetry/opentelemetry-js/blob/main/doc/upgrade-to-2.md)
- [Dynatrace OTLP API docs](https://docs.dynatrace.com/docs/ingest-from/opentelemetry/otlp-api)

--- a/github-copilot-sdk/src/telemetry.ts
+++ b/github-copilot-sdk/src/telemetry.ts
@@ -12,7 +12,7 @@ import { OTLPTraceExporter } from "@opentelemetry/exporter-trace-otlp-proto";
 import { OTLPMetricExporter } from "@opentelemetry/exporter-metrics-otlp-proto";
 import { PeriodicExportingMetricReader, AggregationTemporality } from "@opentelemetry/sdk-metrics";
 import { BatchSpanProcessor } from "@opentelemetry/sdk-trace-base";
-import { Resource } from "@opentelemetry/resources";
+import { resourceFromAttributes } from "@opentelemetry/resources";
 import { ATTR_SERVICE_NAME } from "@opentelemetry/semantic-conventions";
 import { trace, metrics, type Tracer, type Meter } from "@opentelemetry/api";
 
@@ -37,7 +37,7 @@ export function initTelemetry(): void {
   const serviceName = process.env.OTEL_SERVICE_NAME || "copilot-sdk-agent";
   const authHeader = `Api-Token ${otlpToken}`;
 
-  const resource = new Resource({
+  const resource = resourceFromAttributes({
     [ATTR_SERVICE_NAME]: serviceName,
   });
 

